rasa_sdk/knowledge_base/actions.py

```diff
@@ -149,20 +150,38 @@ def _query_objects(
         Returns: list of slots
         """
         object_type = tracker.get_slot(SLOT_OBJECT_TYPE)
-        object_attributes = self.knowledge_base.get_attributes_of_object(object_type)
+        if utils.is_coroutine_action(self.knowledge_base.get_attributes_of_object):
+            object_attributes = await self.knowledge_base.get_attributes_of_object(
+                object_type  # type: ignore
+            )
+        else:
+            object_attributes = self.knowledge_base.get_attributes_of_object(
+                object_type
+            )
 
         # get all set attribute slots of the object type to be able to filter the
         # list of objects
         attributes = get_attribute_slots(tracker, object_attributes)
         # query the knowledge base
-        objects = self.knowledge_base.get_objects(object_type, attributes)
+        if utils.is_coroutine_action(self.knowledge_base.get_objects):
```

Comment:
should be `async` in the interface

Reply:
Same as above, please also update `InMemoryKnowledgeBase`